WFInfo/Services/WarframeProcess/WarframeProcessFinder.cs

```diff
@@ -85,32 +63,64 @@ private void FindProcess()
+                    _identified_process = process;
+                    break;
+                }
+            }
 
-                    _warframe = process;
-
-                    //try and catch any UAC related issues
-                    try
-                    {
-                        bool _ = _warframe.HasExited;
-                    }
-                    catch (System.ComponentModel.Win32Exception e)
-                    {
-                        _warframe = null;
-
-                        Main.AddLog($"Failed to get Warframe process due to: {e.Message}");
-                        Main.StatusUpdate("Restart Warframe without admin privileges, or WFInfo with admin privileges", 1);
+            // Try and catch any UAC related issues
+            if (_identified_process != null)
+            {
+                try
+                {
+                    bool _ = _identified_process.HasExited;
+                }
+                catch (System.ComponentModel.Win32Exception e)
+                {
+                    _identified_process = null;
 
-                    }
-                    OnProcessChanged?.Invoke(_warframe);
-                    return;
+                    Main.AddLog($"Failed to get Warframe process due to: {e.Message}");
+                    Main.StatusUpdate("Restart Warframe without admin privileges, or WFInfo with admin privileges", 1);
+                }
+            }
+            else
+            {
+                if (!_settings.Debug)
+                {
+                    Main.AddLog("Did Not Detect Warframe Process");
+                    Main.StatusUpdate("Unable to Detect Warframe Process", 1);
                 }
             }
 
-            if (!_settings.Debug)
+            // Old warframe process is still cached, lets dispose it and refer to new process
+            if (_warframe != null)
+            {
+                if (_warframe.HasExited)
+                {
+                    Main.dataBase.DisableLogCapture();
+                    _warframe.Dispose();
+                    _warframe = null;
+                }
+            }
+            
+            // New process found? lets refer to it
+            if (_identified_process != null)
             {
-                Main.AddLog("Did Not Detect Warframe Process");
-                Main.StatusUpdate("Unable to Detect Warframe Process", 1);
+                _warframe = _identified_process;
+                if (Main.dataBase.GetSocketAliveStatus())
+                    Debug.WriteLine("Socket was open in verify warframe");
+                Task.Run(async () =>
+                {
+                    await Main.dataBase.SetWebsocketStatus("in game");
+                });
+
+                if (_settings.Auto && !GameIsStreamed)
+                    Main.dataBase.EnableLogCapture();
             }
+
+            // Invoking action
+            OnProcessChanged?.Invoke(_warframe);
```

Comment:
This triggers even if `_warframe` doesn't change, specifically if it was already null and we failed to find a new process.

Reply:
What seems to be correct since I see the same stuff happening earlier. Or do you mean it shouldn't?

Reply:
https://github.com/dimon222/WFinfo/pull/1 was easier to do than describe. Basically, wrap all the change logic into a private setter.